atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueCleaner.java

```diff
@@ -78,12 +86,15 @@ private void cleanNonDedicatedRow(ShardAndStrategy shardStrategy, long partition
     }
 
     private void cleanSweepableTimestamps(ShardAndStrategy shardStrategy, long oldProgress, long newProgress) {
-        if (firstIterationOfSweep(oldProgress)) {
+        if (firstIterationOfSweep(oldProgress) && SweepQueueUtils.tsPartitionCoarse(newProgress + 1) == 0) {
```

Comment:
extract method

Reply:
Yeah, this nasty condition doesn't exist after we refactored the cleaners to use the partition-sets